src/mainClass.js

```diff
@@ -0,0 +1,27 @@
+import { getData, getCategories } from './source';
+
+export class AllItems {
+  constructor(searchCategory, showLinks) {
```

Comment:
Name functions logicly, like showCategories, showLinks. Or not UI specific in model class, onCategoriesUpdate, onLinksUpdated, this is the best.